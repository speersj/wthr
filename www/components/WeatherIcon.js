import React from 'react'
import PropTypes from 'prop-types'
import theme from '../lib/theme'

WeatherIcon.propTypes = { name: PropTypes.string.isRequired }

export default function WeatherIcon({ name }) {
  return (
    <i
      style={{ color: theme.colors[name] || theme.colors.text }}
      data-testid="weather-icon"
      className={iconClassName(name)}
    />
  )
}

function iconClassName(iconName) {
  return `wi wi-forecast-io-${iconName}`
}
