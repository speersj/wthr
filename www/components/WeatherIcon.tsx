import React from "react";
import theme from "../lib/theme";

export default function WeatherIcon({ name }: { name: string }) {
  return (
    <i
      title={name}
      role="img"
      style={{ color: theme.colors[name] || theme.colors.text }}
      className={iconClassName(name)}
    />
  );
}

function iconClassName(iconName: string) {
  return `wi wi-forecast-io-${iconName}`;
}
