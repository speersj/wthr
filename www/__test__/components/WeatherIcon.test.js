import React from 'react'
import { render, cleanup } from 'react-testing-library'
import 'jest-dom/extend-expect'
import WeatherIcon from '../../components/WeatherIcon'

afterEach(cleanup)

describe('WeatherIcon', () => {
  it('outputs an i element with the correct classNames', () => {
    const { getByTestId } = render(<WeatherIcon name="clear-day" />)
    const { className } = getByTestId('weather-icon')
    expect(className).toEqual('wi wi-forecast-io-clear-day')
  })

  it('outputs correct html', () => {
    const { getByTestId } = render(<WeatherIcon name="rain" />)
    expect(getByTestId('weather-icon')).toMatchSnapshot()
  })
})
