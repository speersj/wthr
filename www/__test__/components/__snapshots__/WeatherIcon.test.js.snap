// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherIcon outputs correct html 1`] = `
<i
  class="wi wi-forecast-io-rain"
  data-testid="weather-icon"
  style="color: rgb(139, 233, 253);"
/>
`;
